common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -123,10 +120,6 @@ private String checkNotNullAndNotEmpty(String name) {
         return name;
     }
 
-    protected abstract T newConstant(int id, String name);
+    protected abstract T newConstant(String name);
 
-    @Deprecated
-    public final int nextId() {
```

Comment:
this is an API breaking change right? It was brought back in 4.1 bcz we broke the API at some point https://github.com/netty/netty/commit/b4b14ea19f343fbe7a0993a92856d573683de810